data/ssystem_minor.ini

```diff
@@ -947,7 +1046,8 @@ coord_func                     = comet_orbit
 dust_brightnessfactor          = 1.5
 dust_lengthfactor              = 0.4
 dust_widthfactor               = 1.5
-name                           = 2I/Borisov [C/2019 Q4 (Borisov)]
+name                           = 2I/Borisov
+provisional_designation        = C/2019 Q4
```

Comment:
The C/2019 Q4 is not a provisional code. It is the current IAU style of a unique designation of a comet. 
See your list periodic_comet_codes.fab: What we know as 1P/Halley has indeed the discovery-based code "C/1682 Q1", although it has never been observed under this code (ok, maybe with 4m-class instruments :-) According to https://en.wikipedia.org/wiki/Halley%27s_Comet, this designation should actually change to indicate which perihel run is meant.

When I started in the field, there were provisional designations like  "1982i (Halley)". These year-small-letter codes were clearly "year/sequence of (re)discovery", and when identity with a known comet was confirmed, the known name was added. E.g., when the first comet of 1910 was discovered, it was labelled 1910a (Not sure if the YYYYx labelling was even used in the early 20th ct. or this was the label used in the early 1990s). After perihel (or sometimes later?), the final name was given as "1910 I (Johannesburg)". (Generally. YYYY of perihel and Roman sequence numbers, making it clear which apparition was discussed for a periodic comet) And now it is labelled "C/1910 A1 Great January Comet". 

Halley can be identified with e.g. 1P/1909 R1, 1910 II, 1909c and 1P/1982 U1, 1986 III, 1982i (see WP). The first labels are the currently favoured style of discovery year/half-month. The second was the traditional numerical order of perihel dates. Third the provisional discovery code. 

Maybe rename this entry to "date_code". This would not necessarily mean it was labelled like this just after discovery (this would be 1982i for Halley). provisional_designation for me would read as "label used when it was first discovered and news were circulated." This could be an optional entry when labelling style has been changed. I know, this is a bit messy. If we want to show all flavours (which I would like to have!), Infotext name could look like

1P/Halley (P/1982 U1 - 1982i - 1986 III)

For this, we'd need additional optional entries "perihel_code=1986 III" and "discovery_code=1982i".

So, to sum it up, please extend this list to 

date_code ... current date-based code like "P/1982 U1" or (for Steins above) "1969 VC"
perihel_code ... used for comets during most of the 19/20th centuries, e.g. "1986 III" 
discovery_code ... provisional labelling made immediately after discovery, e.g. "1982i".

Reply:
I use `iau_designation` field for minor planets and comets now. I think this is enough for current task :)

Reply:
I'll use data from Kronk's _Cometography_ to add discovered/recovered dates and _perihel_code_/_discovery_code_ for comets in our `ssystem_minor.ini` file. Probably on next iteration we should update `ssystem_1000comets.ini` file by data from _Cometography_.